File: core/src/main/java/me/pepperjackdev/chess/core/data/state/placement/Placement.java
package me.pepperjackdev.chess.core.data.state.placement;

import me.pepperjackdev.chess.core.piece.Piece;
import me.pepperjackdev.chess.core.position.Position;

import java.util.Optional;

public interface Placement {
    Size getSize();
    Optional<Piece> getPiece(Position position);
}

File: src/chess/placement.c
#include "chess/placement.h"
#include "chess/utils/tuple.h"

Piece placement_get_piece(Placement *placement, Tuple2 position) {
    return (*placement)[t2toi(position)];
}

void placement_set_piece(Placement *placement, Tuple2 position, Piece piece) {
    (*placement)[t2toi(position)] = piece;
}